src/sysupdate/sysupdate-resource.c

```diff
@@ -132,16 +119,42 @@ static int resource_load_from_directory(
                         return log_error_errno(errno, "Failed to stat %s/%s: %m", rr->path, de->d_name);
                 }
 
-                if ((st.st_mode & S_IFMT) != m)
+                if (!(S_ISDIR(st.st_mode) && S_ISREG(m)) && ((st.st_mode & S_IFMT) != m))
                         continue;
 
-                r = pattern_match_many(rr->patterns, de->d_name, &extracted_fields);
-                if (r < 0)
+                if (relpath != NULL)
+                        rel_joined = path_join(relpath, de->d_name);
+                else
+                        rel_joined = strdup(de->d_name);
```

Comment:
path_join() happily skips over NULL parameters btw, hence you can just drop this conditionalization, and call path_join() in all cases